src/couch/src/couch_server.erl

```diff
@@ -114,6 +117,35 @@ create(DbName, Options0) ->
 delete(DbName, Options) ->
     gen_server:call(couch_server, {delete, DbName, Options}, infinity).
 
+
+exists(DbName) ->
+    RootDir = config:get("couchdb", "database_dir", "."),
+    Engines = get_configured_engines(),
+    Possible = lists:foldl(fun({Extension, Engine}, Acc) ->
```

Comment:
We have exactly same lines in `get_engine(Server, DbName)` function. We can create a common function such as:
- `select_candidate_engines(DbName, RootDir, Engines)`
- `select_possible_engines(DbName, RootDir, Engines)`